lock-api/src/main/java/com/palantir/lock/client/TimelockRpcClient.java

```diff
@@ -0,0 +1,149 @@
+
+    @POST
+    @Path("fresh-timestamp")
+    long getFreshTimestamp();
+
+    @POST
+    @Path("fresh-timestamps")
+    TimestampRange getFreshTimestamps(@Safe @QueryParam("number") int numTimestampsRequested);
+
+    @POST
+    @Path("lock-immutable-timestamp")
+        // TODO (jkong): Can this be deprecated? Are there users outside of Atlas transactions?
+    LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#startAtlasDbTransactionV3(
+     * StartIdentifiedAtlasDbTransactionRequest)} instead; ignore the partition information if it is not useful for you.
+     */
+    @POST
+    @Path("start-atlasdb-transaction")
+    @Deprecated
+    StartAtlasDbTransactionResponse startAtlasDbTransaction(IdentifiedTimeLockRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#startAtlasDbTransactionV3(
+     * StartIdentifiedAtlasDbTransactionRequest)} instead.
+     */
+    @POST
+    @Path("start-identified-atlasdb-transaction")
+    @Deprecated
+    StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction(
+            StartIdentifiedAtlasDbTransactionRequest request);
+
+    @POST
+    @Path("immutable-timestamp")
+    long getImmutableTimestamp();
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#lockV2(LockRequest) instead}.
+     */
+    @POST
+    @Path("lock")
+    @Deprecated
+    LockResponse lock(LockRequest request);
+
+    @POST
+    @Path("await-locks")
+    WaitForLocksResponse waitForLocks(WaitForLocksRequest request);
+
+    /**
+     * @deprecated Please use {@link TimelockRpcClient#refreshLockLeasesV2(Set)} instead}.
+     */
+    @POST
+    @Path("refresh-locks")
+    @Deprecated
+    Set<LockToken> refreshLockLeases(Set<LockToken> tokens);
+
+    @POST
+    @Path("refresh-locks-v2")
+    LeasableRefreshLockResponse refreshLockLeasesV2(Set<LockToken> tokens);
+
+    @POST
+    @Path("lock-v2")
+    LeasableLockResponse lockV2(LockRequest request);
+
+    @POST
+    @Path("start-atlasdb-transaction-v3")
+    LeasableStartIdentifiedAtlasDbTransactionResponse startAtlasDbTransactionV3(
+            StartIdentifiedAtlasDbTransactionRequest request);
+
+    /**
+     * Releases locks associated with the set of {@link LockToken}s provided.
+     * The set of tokens returned are the tokens for which the associated locks were unlocked in this call.
+     * It is possible that a token that was provided is NOT in the returned set (e.g. if it expired).
+     * However, in this case it is guaranteed that that token is no longer valid.
+     *
+     * @param tokens Tokens for which associated locks should be unlocked.
+     * @return Tokens for which associated locks were unlocked
+     */
+    @POST
+    @Path("unlock")
+    Set<LockToken> unlock(Set<LockToken> tokens);
+
+    @GET
+    @Path("leader-time")
+    IdentifiedTime getLeaderTime();
```

Comment:
I would probably not add the new endpoints in this PR, because they won't be implemented on the server and so will lead to slightly weird code for people in the meantime?